### PR TITLE
Create /stories endpoint with list of story cards

### PR DESCRIPTION
**Pull request by e-wai**
## Notion ticket link

[Create a large table of story cards](https://www.notion.so/uwblueprintexecs/a658c933d18c48d4b43fd99bda2bd3b4?v=4a47edc475b34bb99b25ff1fae2a5507&p=f2ebfec6aceb419492796d9fd1dceff7)

## Implementation description
* Create /stories route and add button to access on default screen. Remove sample `StoryCard` that was there
* Create stories page with `StoryList` component and appropriate GraphQL query. This was based on the starter code entity query
* Fix up some CSS so it looks better

## Steps to test
1. Start with empty stories database (or you can do this at the end if you prefer). Go to /stories endpoint either manually or via button on main page
2. Populate your stories database with data. There's no error handling if the youtube link isn't real so pls make it real
3. Login and go to the /stories endpoint. View stories; be happy.

I tried with stories in the below GIF and an empty array and both seemed ok? I'm considering opening a PR to extract the MainPageButton into its own component and to add it to the /stories page. The code is pretty much duplicated across all the non-default pages right now, thoughts?

Tada 🎉 
React App - Google Chrome 2021-06-13 23-55-36 (2)

* Should I be doing something about this

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] For backend changes, I have run the appropriate linters:  `docker exec -it <backend-container-id> /bin/bash -c "black . && isort --profile black ."`
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/pages/StoriesPage.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      id
+      title
+      description
+      youtubeLink
+      level
+    }
+  }
+`;
+
+const StoryList = () => {
```
Comment (gaoxk) [inline frontend/src/components/pages/StoriesPage.tsx:17]: Can you make StoryList its own component in `/story` and pass in `stories` as prop? Let's make it as stateless as possible. 

And, can you change this file & route from `StoriesPage` to `Homepage`? ;)

File: frontend/src/App.tsx [hunk 2]
```diff
@@ -33,6 +34,7 @@ const App = () => {
           <PrivateRoute exact path="/entity/create" component={CreatePage} />
           <PrivateRoute exact path="/entity/update" component={UpdatePage} />
           <PrivateRoute exact path="/entity" component={DisplayPage} />
+          <PrivateRoute exact path="/stories" component={StoriesPage} />
```
Comment (gaoxk) [inline frontend/src/App.tsx:37]: ditto comment about rename to homepage

File: frontend/src/components/pages/Default.tsx [hunk 3]
```diff
@@ -38,21 +37,14 @@ const GetButton = () => {
...
-      It wasn't. Had it been a hundred? It quite possibly could have been. 
-      Two hundred? Probably not. His mind wandered until he caught himself 
-      and again tried to imagine it was the first time.`,
-      youtubeLink: "https://www.youtube.com/watch?v=_OBlgSz8sSM",
-      level: 3,
-      translatedLanguages: [],
-      contents: [],
-    },
-  },
+const StoriesButton = () => {
```
Comment (gaoxk) [inline frontend/src/components/pages/Default.tsx:40]: ditto comment about rename to homepage

**Review by gaoxk — CHANGES_REQUESTED** [review]
> There's no error handling if the youtube link isn't real

Hmm. Try [regex for youtube urls](https://irishdotnet.dev/how-to-validate-a-youtube-url-using-javascript). It could be fooled, but the user would have to be malicious, which is unlikely since only planet read employees will generate story data

File: frontend/src/components/pages/StoriesPage.tsx (hunk 1 above)
Comment (e-wai) [reply frontend/src/components/pages/StoriesPage.tsx:17]: Sounds good - what's the benefit of making `StoryList` more stateless?

**Comment by e-wai** [comment]
Where would we want to put this youtube link validation? Is this something for the backend to raise as an error when the `createStory` mutation is called with the link, or for us to know when to display some placeholder thumbnail instead of trying to load the iframe?

**Comment by gaoxk** [comment]
Backend is a good idea if there were a quick way to verify youtube link validity. Based on a (quick) googe search, there doesnt seem to be a convenient api exposed by youtube. For simplicities sake, we can do a regex validation on youtube links once the UI is built out for admins to enter story info (that'll be next term)

**Comment by e-wai** [comment]
Opened an issue for the Youtube link validation here -> https://www.notion.so/uwblueprintexecs/Youtube-link-validation-1f1cbeafb8c2426299e57e3f45332787

File: frontend/src/components/story/StoryCard.css [hunk 4]
```diff
@@ -14,9 +15,7 @@
 
 .story-card-details {
     flex-basis: 40%;
-    padding-bottom: 17px;
-    padding-left: 26px;
-    padding-top: 17px;
+    padding: 17px 0px 17px 26px;
```
Comment (gaoxk) [inline frontend/src/components/story/StoryCard.css:18]: nice!

**Review by gaoxk — APPROVED** [review]
🚢

**Pull request merged**